### DISCUSSION TITLE
Argument naming for the same feature for different functions.

### DISCUSSION THREAD

**Discussion by dememax** [mesonbuild/meson#13955] [General]
Why do different meson functions like `executable()` and `test()` have the same arguments _(such as dependencies)_ with different names (e.g., **dependencies** vs **depends**)?

**Comment by eli-schwartz** [comment]
- Dependencies are libraries and headers that are needed for compiling and linking.
- Depends are build ordering rules (A has to be built before B).

The former can sometimes imply the latter, if a library is also a target which is built rather than being looked up on the system.
